projects/js-packages/ai-client/src/hooks/use-image-generator/index.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+import debugFactory from 'debug';
+/**
+ * Internal dependencies
+ */
+import requestJwt from '../../jwt/index.js';
+
+const debug = debugFactory( 'ai-client:use-image-generator' );
+
+const useImageGenerator = () => {
+	const generateImage = async function (): Promise< { data: Array< { url: string } > } > {
+		let token = '';
+
+		try {
+			token = ( await requestJwt() ).token;
+		} catch ( error ) {
+			debug( 'Error getting token: %o', error );
+			return Promise.reject( error );
+		}
+
+		try {
+			debug( 'Generating image' );
+
+			// TODO: Find a proper prompt for the image generation
+			const imageGenerationPrompt = ``;
+
+			const URL = 'https://public-api.wordpress.com/wpcom/v2/jetpack-ai-image';
+
+			const body = {
+				prompt: imageGenerationPrompt,
+				feature: 'jetpack-ai-image-generator',
```

Comment:
```suggestion
				feature: 'jetpack-ai-featured-image-generator',
```

Reply:
I see this is in the ai-client, so not specific for the featured image. We may need to have the feature as a parameter with `jetpack-ai-image-generator` as default and then use `jetpack-ai-featured-image-generator` when calling this function in the modal later.